### PR TITLE
[1LP][RFR] uncollect ansible tests for 5.8 pod appliance

### PR DESCRIPTION
**Pull request by izapolsk**
1. 5.8 pod appliances don't support embedded ansible.
2. updated uncollects a bit
3. fixed some tests

### CODE REVIEW

File: cfme/tests/ansible/test_embedded_ansible_crud.py [hunk 1]
```diff
@@ -1,8 +1,14 @@
 import pytest
 
-from cfme.utils.version import current_version
 from cfme.utils.wait import wait_for
 
+pytestmark = [
+    pytest.mark.uncollectif(lambda appliance: appliance.version < "5.9" and appliance.is_pod,
+                            reason="5.8 pod appliance doesn't support embedded ansible"),
+    pytest.mark.uncollectif(lambda appliance: appliance.version < "5.8"),
```
Comment (jamesooden) [inline cfme/tests/ansible/test_embedded_ansible_crud.py]: **QUESTION:** Should we have a reason on this one?

Comment (izapolsk) [reply cfme/tests/ansible/test_embedded_ansible_crud.py]: we definitely should.  I suspect ansible support  was added in 5.8.  But I'm not 100% sure.  I'd ask test owners.

Comment (izapolsk) [reply cfme/tests/ansible/test_embedded_ansible_crud.py]: @lcouzens, could you confirm whether I'm right ? btw, would you like to add reason everywhere in ansible tests?

**Comment by jamesooden** [comment]
**QUESTION:**  Are we expecting this from PRT:

cfme/tests/ansible/test_embedded_ansible_services.py::test_service_ansible_playbook_order_credentials Exception TypeError: "argument of type 'NoneType' is not iterable" in <generator object pytest_runtest_call at 0x7f02184c62d0> ignored
Exception AttributeError: AttributeError('catch_log_handler',) in <generator object pytest_runtest_call at 0x7f02184c6460> ignored
Exception TypeError: "argument of type 'NoneType' is not iterable" in <generator object pytest_runtest_call at 0x7f0209912190> ignored
Exception AttributeError: AttributeError('catch_log_handler',) in <generator object pytest_runtest_call at 0x7f0218453410> ignored

**Comment by jamesooden** [comment]
Oh, my just look at your PRT results.   It's not your change but there are errors in the ansible playbook tests.

File: cfme/tests/ansible/test_embedded_ansible_crud.py (hunk 1 above)
Comment (lcouzens) [reply cfme/tests/ansible/test_embedded_ansible_crud.py]: Yes embedded ansible was first added in 5.8, as for reasons I don't mind.

File: cfme/tests/ansible/test_embedded_ansible_basic.py [hunk 2]
```diff
@@ -5,15 +5,16 @@
...
 from cfme.utils.appliance.implementations.ui import navigate_to
 from cfme.utils.update import update
-from cfme.utils.version import current_version
 from cfme.utils.wait import wait_for
 
 pytestmark = [
     pytest.mark.long_running,
     pytest.mark.meta(server_roles=["+embedded_ansible"]),
-    pytest.mark.uncollectif(lambda: current_version() < "5.8"),
+    pytest.mark.uncollectif(lambda appliance: appliance.version < "5.8"),
```
Comment (rlbabyuk) [inline cfme/tests/ansible/test_embedded_ansible_basic.py]: Required: Please add a reason info for this uncollectif, also update the same places below

**Comment by izapolsk** [comment]
@jamesooden, this isn't related to my changes. this is something common. I'll try to fix it in another PR.

File: cfme/tests/ansible/test_embedded_ansible_basic.py [hunk 3]
```diff
@@ -5,15 +5,17 @@
...
 from cfme.utils.appliance.implementations.ui import navigate_to
 from cfme.utils.update import update
-from cfme.utils.version import current_version
 from cfme.utils.wait import wait_for
 
 pytestmark = [
     pytest.mark.long_running,
     pytest.mark.meta(server_roles=["+embedded_ansible"]),
-    pytest.mark.uncollectif(lambda: current_version() < "5.8"),
+    pytest.mark.uncollectif(lambda appliance: appliance.version < "5.8",
```
[the remark this answers is not in the document]
Comment (izapolsk) [reply cfme/tests/ansible/test_embedded_ansible_basic.py]: I don't mind. I just slightly updated existing calls by moving them to global pytestmark or removing deprecated current_version.

**Pull request merged**